pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -205,27 +222,32 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 		{Name: "ROOK_IS_DEVICE", Value: "true"},
 	}...)
 
+	var args []string
 	// If the OSD runs on PVC
 	if osdProps.onPVC() {
 		// add the PVC size to the pod spec so that if the size changes the OSD will be restarted and pick up the change
 		envVars = append(envVars, v1.EnvVar{Name: "ROOK_OSD_PVC_SIZE", Value: osdProps.pvcSize})
 
-		// Append tuning flag if necessary
+		// Append slow tuning flag if necessary
 		if osdProps.tuneSlowDeviceClass {
-			err := c.osdRunFlagTuningOnPVC(osd.ID)
+			err := c.osdRunFlagSlowTuningOnPVC(osd.ID)
 			if err != nil {
-				return nil, errors.Wrapf(err, "failed to apply tuning on osd %q", strconv.Itoa(osd.ID))
+				return nil, errors.Wrapf(err, "failed to apply slow tuning on osd %q", strconv.Itoa(osd.ID))
+			}
+		} else if osdProps.tuneFastDeviceClass { // Append fast tuning flag if necessary
+			for flag, val := range defaultTuneFastSettings {
+				args = append(args, opconfig.NewFlag(flag, val))
 			}
 		}
 	}
 
 	var command []string
-	var args []string
+
 	// If the OSD was prepared with ceph-volume and running on PVC and using the LVM mode
 	if osdProps.onPVC() && osd.CVMode == "lvm" {
 		// if the osd was provisioned by ceph-volume, we need to launch it with rook as the parent process
 		command = []string{path.Join(rookBinariesMountPath, "tini")}
-		args = []string{
+		args = append(args,
```

Comment:
We need to append the tuning args at the end, rather than prepend. Otherwise, in this case the new args will be before the "ceph osd start" command.

Reply:
Done.